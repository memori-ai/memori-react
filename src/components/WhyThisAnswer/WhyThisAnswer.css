p.memori--whythisanswer-no-results {
    padding: 0.5rem 1rem;
    border: 1px solid var(--memori-warning-color);
    border-radius: var(--memori-border-radius);
    margin: 0.5rem 0;
    background: #fff;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
    color: var(--memori-warning-color);
    text-align: center;
}

.memori--whythisanswer-question-answered {
    padding: 1rem;
    border-radius: var(--memori-border-radius);
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.25);
}

ul.memori--whythisanswer-list {
    padding: 0;
    margin: 0;
    list-style: none;
}

ul.memori--whythisanswer-list li {
    padding: 0.5rem 0;
    border-bottom: 1px solid #edf0f3;
    margin: 0.5rem 0;
}

ul.memori--whythisanswer-list li:last-child {
    border-bottom: none;
}

.memori--whythisanswer-confidence {
    display: inline-block;
    padding: 0.25rem 0.5rem;
    border: 1px solid var(--memori-primary);
    border-radius: var(--memori-border-radius);
    margin-right: 1em;
    background: #fff;
    color: var(--memori-primary);
    font-size: 0.75rem;
}

.memori--whythisanswer-title {
    display: flex;
    align-items: center;
    margin-bottom: 1rem;
}

@media (max-width: 768px) {
    .memori--whythisanswer-title {
        flex-direction: column;
        align-items: flex-start;
    }

    .memori--whythisanswer-confidence {
        margin-right: 0;
        margin-bottom: 0.5rem;
    }
}

.memori--whythisanswer-title-text p {
    margin: 0;
}

.memori-whythisanswer-drawer .memori-media-items--grid {
    display: none;
}

.memori-whythisanswer-drawer .memori-snippet--content {
    font-size: 0.75rem;
}
