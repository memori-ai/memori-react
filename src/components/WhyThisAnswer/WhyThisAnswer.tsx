import {
  SearchMatches,
  Message,
} from '@memori.ai/memori-api-client/dist/types';
import { useCallback, useEffect, useState } from 'react';
import memoriApiClient from '@memori.ai/memori-api-client';
import Drawer from '../ui/Drawer';
import Spin from '../ui/Spin';
import Expandable from '../ui/Expandable';
import toast from 'react-hot-toast';
import { getErrori18nKey } from '../../helpers/error';
import { useTranslation } from 'react-i18next';
import Snippet from '../Snippet/Snippet';
import MediaWidget from '../MediaWidget/MediaWidget';

export interface Props {
  apiURL: string;
  sessionID: string;
  message: Message;
  initialMatches?: SearchMatches[];
  visible?: boolean;
  closeDrawer: () => void;
  _TEST_loading?: boolean;
}

const addQuestionMark = (question: string) =>
  question.endsWith('?') ? question : `${question}?`;

const WhyThisAnswer = ({
  message,
  apiURL,
  sessionID,
  visible = true,
  initialMatches = [],
  closeDrawer,
  _TEST_loading = false,
}: Props) => {
  const { t } = useTranslation();

  const client = memoriApiClient(apiURL);
  const searchMemory = client.search.searchMemory;

  const [matches, setMatches] = useState<SearchMatches[]>(initialMatches);
  const [loading, setLoading] = useState(_TEST_loading);

  /**
   * Fetch matching memories
   */
  const fetchMemories = useCallback(async () => {
    setLoading(true);

    if (_TEST_loading) return;

    try {
      const { matches, ...response } = await searchMemory(sessionID, {
        searchType: 'Semantic',
        numberOfResults: 3,
        text: message.questionAnswered,
        date: message.date,
        placeName: message.placeName,
        placeLatitude: message.placeLatitude,
        placeLongitude: message.placeLongitude,
        placeUncertaintyKm: message.placeUncertaintyKm,
        contextVars: message.contextVars,
        tag: message.tag,
        memoryTags: message.memoryTags,
      });

      if (response.resultCode !== 0) {
        console.error(response);
        toast.error(t(getErrori18nKey(response.resultCode)));
      } else {
        setMatches(matches ?? []);
      }
    } catch (err) {
      console.error('WHYTHISANSWER/FETCH', err);
      setMatches(initialMatches ?? []);
    }

    setLoading(false);
  }, [message, sessionID]);
  useEffect(() => {
    fetchMemories();
  }, [fetchMemories, message, sessionID]);

  return (
    <Drawer
      open={visible}
      width="80%"
      animated={false}
      className="memori-whythisanswer-drawer"
      onClose={() => closeDrawer()}
      title={t('whyThisAnswer')}
    >
      <p>{t('whyThisAnswerHelper')}</p>

      {message.questionAnswered && (
        <p className="memori--whythisanswer-question-answered">
          <strong>{t('question') || 'Question'}:</strong>{' '}
          {message.questionAnswered}
        </p>
      )}

      <Spin primary spinning={loading}>
        {!loading && matches.length === 0 && (
          <p role="info" className="memori--whythisanswer-no-results">
            {t('nothingFound')}
          </p>
        )}
        {loading && matches.length === 0 && (
          <ul className="memori--whythisanswer-list memori--whythisanswer-skeleton">
            <li>
              <div className="memori--whythisanswer-title">
                <span className="memori--whythisanswer-confidence">
                  <span className="memori--whythisanswer-skeleton-text"></span>
                </span>
                <div className="memori--whythisanswer-title-text">
                  <p className="memori--whythisanswer-skeleton-text"></p>
                </div>
              </div>
              <p>
                <div className="memori--whythisanswer-skeleton-text"></div>
                <div className="memori--whythisanswer-skeleton-text"></div>
                <div className="memori--whythisanswer-skeleton-text"></div>
              </p>
              <div className="memori--whythisanswer-skeleton-block"></div>
            </li>
            <li>
              <div className="memori--whythisanswer-title">
                <span className="memori--whythisanswer-confidence">
                  <span className="memori--whythisanswer-skeleton-text"></span>
                </span>
                <div className="memori--whythisanswer-title-text">
                  <p className="memori--whythisanswer-skeleton-text"></p>
                  <p className="memori--whythisanswer-skeleton-text"></p>
                </div>
              </div>
              <p>
                <div className="memori--whythisanswer-skeleton-text"></div>
                <div className="memori--whythisanswer-skeleton-text"></div>
              </p>
              <div className="memori--whythisanswer-skeleton-block"></div>
            </li>
          </ul>
        )}
        {matches.length > 0 && (
          <ul className="memori--whythisanswer-list">
            {matches.map(m => (
              <li key={m.memory.memoryID}>
                <div className="memori--whythisanswer-title">
                  <span className="memori--whythisanswer-confidence">
                    {m.confidenceLevel}
                  </span>
                  <div className="memori--whythisanswer-title-text">
                    <p>
                      <strong>{addQuestionMark(m.memory.title ?? '')}</strong>
                    </p>
                    <p>
                      {m.memory.titleVariants
                        ?.map(t => addQuestionMark(t))
                        ?.join(' | ')}
                    </p>
                  </div>
                </div>
                {m.memory.answers?.map((a, i) => (
                  <p key={i} className="memori--whythisanswer-answer">
                    <Expandable rows={3}>{a.text}</Expandable>
                  </p>
                ))}

                <MediaWidget
                  links={m.memory.media?.filter(
                    m => m.mimeType === 'text/html'
                  )}
                />

                {m.memory.media
                  ?.filter(m => m.mimeType === 'text/plain')
                  ?.map(m => (
                    <Expandable
                      rows={2}
                      key={m.mediumID}
                      lineHeightMultiplier={2}
                      innerClassName="memori--whythisanswer-snippet-expandable"
                    >
                      <Snippet
                        key={m.mediumID}
                        medium={m}
                        showCopyButton={false}
                      />
                    </Expandable>
                  ))}
              </li>
            ))}
          </ul>
        )}
      </Spin>
    </Drawer>
  );
};

export default WhyThisAnswer;
