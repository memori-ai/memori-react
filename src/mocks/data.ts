import {
  Medium,
  Memori,
  Message,
  Memory,
  Integration,
  Tenant,
  DialogState,
  KnownFact,
  ExpertReference,
  User,
  Venue,
} from '@memori.ai/memori-api-client/dist/types';

export const sessionID = '131165be-9d1a-42fb-a3ce-e8f86d40c88f';

export const tenant: Tenant = {
  adminCount: 3,
  config: {
    name: 'AIsuru',
    requirePosition: false,
    showNewUser: true,
  },
  creationTimestamp: '2023-05-31T14:32:48.885287Z',
  description: 'AIsuru',
  disableRegistration: false,
  id: 'www.aisuru.com',
  lastChangeTimestamp: '2023-05-31T14:32:48.885287Z',
  logoURL: 'https://aisuru.com/images/aisuru/logo.png',
  maxAdmins: 0,
  maxCompletions: 0,
  maxCompletionsPerUser: 0,
  maxFreeSessions: 400,
  maxFreeSessionsPerUser: 100,
  maxMemoriPerAdmin: 0,
  maxMemoriPerUser: 3,
  maxTotalMemori: 0,
  maxUsers: 0,
  memoriCount: 0,
  name: 'www.aisuru.com',
  nonFreeSessionCost: 0.02,
  paying: true,
  tenantID: '96caa4b4-31a4-48e5-8163-dec61869a2a7',
  theme: 'aisuru',
  userCount: 0,
};

export const user: User = {
  tenant: 'localhost:3000',
  userID: '97c42d18-ffe4-47e1-a3c7-e42729f1e6a3',
  userName: 'nzambello',
  eMail: 'nicola@nzambello.dev',
  admin: false,
  maxMemori: 0,
  enableMemoriCreation: true,
  enableBoardOfExperts: true,
  maxFreeSessions: 0,
  tnCAndPPAccepted: true,
  tnCAndPPAcceptanceDate: '2021-03-01T00:00:00.000Z',
  pAndCUAccepted: true,
  pAndCUAcceptanceDate: '2021-03-01T00:00:00.000Z',
  birthDate: '1900-03-01T00:00:00.000Z',
  age: 28,
  avatarURL: 'https://avatars.githubusercontent.com/u/21101435?v=4',
};

export const memori: Memori = {
  memoriID: '25ced51c-3520-41af-8bbe-222d861b8e32',
  engineMemoriID: '66b4e161-2431-4b21-9b70-d8c27de730ca',
  name: 'Memori',
  memoriConfigurationID: 'MemoriCloud-it_IT',
  description: 'Lorem ipsum.',
  voiceType: 'male',
  isGiver: true,
  isReceiver: false,
  privacyType: 'PUBLIC',
  needsPosition: false,
  culture: 'it-IT',
  categories: [],
  publishedInTheMetaverse: true,
  exposed: true,
  enableCompletions: true,
  nsfw: false,
  ageRestriction: 14,
  contentQualityIndex: 66.6,
  contentQualityIndexTimestamp: '2021-03-01T12:00:00.000Z',
  ownerUserName: 'username',
  ownerTenantName: 'aisuru.com',
  ownerUserID: '97c42d18-ffe4-47e1-a3c7-e42729f1e6a3',
  metaverseEnvironment: 'synthwave',
  secretToken: 'awanagana',
  giverPIN: '1234',
  giverTag: 'üßë‚Äçüíª',
  avatarURL: 'https://aisuru.com/images/aisuru/square_logo.png',
  coverURL: 'https://aisuru.com/images/aisuru/og-image.png',
};

export const history: Message[] = [
  {
    text: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Prova',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: "Mi dispiace, le mie risposte sono limitate. Devi farmi le domande giuste. C'√® altro che vuoi sapere?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Come faccio a fare delle cose con questa cosa?',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ecco qui come.',
    media: [
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
        url: 'https://memori.ai',
        mimeType: 'text/html',
        title: 'Link',
      },
    ],
    timestamp: '2021-03-01T12:00:00.000Z',
    contextVars: {
      TEST: 'test',
    },
  },
  {
    text: 'Ah, grazie! Ciao!',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Arrivederci.',
    timestamp: '2021-03-01T12:00:00.000Z',
    contextVars: {
      TEST: 'test',
    },
  },
];

export const historyWithMedia: Message[] = [
  {
    text: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Prova',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: "Mi dispiace, le mie risposte sono limitate. Devi farmi le domande giuste. C'√® altro che vuoi sapere?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Come faccio a fare delle cose con questa cosa?',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ecco qui delle cose per te.',
    media: [
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
        url: 'https://memori.ai/en',
        mimeType: 'text/html',
        title: 'Link Memori Srl',
      },
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efce',
        url: 'https://rawmaterial.it/en',
        mimeType: 'text/html',
        title: 'Link RawMaterial',
      },
      {
        mediumID: '95226d7e-7bae-465e-8b80-995587bb5971',
        mimeType: 'text/html',
        title: 'Introducing Plone Remix | Vimeo',
        url: 'https://vimeo.com/766468314',
      },
      {
        mediumID: '95226d7e-7bae-465e-8b80-995587bb5969',
        mimeType: 'text/html',
        title: 'A sustainable web: is it possible? - Nicola Zambello | YouTube',
        url: 'https://www.youtube.com/watch?v=feH26j3rBz8',
      },
      ...Array.from({ length: 3 }, (_, i) => ({
        mediumID: `95226d7e-7bae-465e-8b80-995587bb597${i}`,
        mimeType: 'image/png',
        title: `Image ${i}`,
        url: `https://picsum.photos/${i % 2 ? '200' : '300'}/${
          i % 3 ? '300' : '200'
        }?random=${i}`,
      })),
    ],
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ah, grazie! Ciao!',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Arrivederci.',
    timestamp: '2021-03-01T12:00:00.000Z',
  },
];

export const historyWithAIGeneratedMessages: Message[] = [
  {
    text: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Prova',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Sa. Sa. Prova',
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Come faccio a fare delle cose con questa cosa?',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ecco qui come.',
    media: [
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
        url: 'https://memori.ai',
        mimeType: 'text/html',
        title: 'Link',
      },
    ],
    timestamp: '2021-03-01T12:00:00.000Z',
    generatedByAI: true,
    contextVars: {
      TEST: 'test',
    },
  },
  {
    text: 'Ah, grazie! Ciao!',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Arrivederci.',
    timestamp: '2021-03-01T12:00:00.000Z',
    contextVars: {
      TEST: 'test',
    },
  },
];

export const dialogState: DialogState = {
  state: 'R1',
  previousState: 'I0',
  stateName: 'WaitingForReceiverQuestion',
  confidence: 1,
  knownTags: {
    '‚ò†Ô∏è': 'test',
    'üòé': 'Ciccio',
  },
  emission: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
  hints: ['Va bene', 'No grazie'],
  media: [],
  acceptsTimeout: true,
  acceptsAbort: false,
  acceptsMedia: false,
  acceptsDate: false,
  acceptsPlace: false,
  acceptsTag: false,
  giverID: 'c832e2dc-403c-4baf-a3b7-2374e100dbcf',
  contextVars: {},
};

export const medium: Medium = {
  mediumID: 'a196b513-d745-4121-8913-8f457b999add',
  url: 'https://api.lorem.space/image/game?w=150&h=220&hash=8B7BCDC2',
  mimeType: 'image/jpeg',
  title: 'Game Cover',
  creationTimestamp: '2022-05-03T08:57:30.584439Z',
  creationName: 'fcfda4a7-bb19-41eb-8384-db0d22f825ec',
  lastChangeTimestamp: '2022-05-03T14:19:09.932865Z',
  lastChangeName: 'fcfda4a7-bb19-41eb-8384-db0d22f825ec',
};

export const integration: Integration = {
  integrationID: 'cb3c4776-7f0b-4f97-a773-c32a5d7a3bf1',
  memoriID: '25ced51c-3520-41af-8bbe-222d861b8e32',
  type: 'LANDING_EXPERIENCE',
  state: 'NEW',
  deviceEmails: [],
  customData:
    '{"textColor":"#2a2a2a","buttonBgColor":"#823ce1","buttonTextColor":"#ffffff","name":"Web","globalBackground":"https://assets.memori.ai/api/v2/asset/364e498c-11da-42d5-9e32-19e5d137d4b8.jpeg","blurBackground":true,"innerBgColor":"light","innerBgAlpha":0.8,"multilanguage":true,"avatar":"readyplayerme","avatarURL":"https://assets.memori.ai/api/v2/asset/b791f77c-1a94-4272-829e-eca82fcc62b7.glb#1669663599444"}',
  resources: [],
  publish: true,
  creationTimestamp: '2022-06-11T14:13:45.685038Z',
  lastChangeTimestamp: '2022-06-11T14:13:45.685038Z',
};

export const knownFact: KnownFact = {
  knownFactID: 'b0b0b0b0-b0b0-b0b0-b0b0-b0b0b0b0b0b3',
  knownFactType: 'ShortTerm',
  text: 'I am a known fact. Quisque in ultrices lectus. Nulla at urna diam. Proin sodales lobortis libero eu facilisis.',
  creationTimestamp: '2023-12-01T13:40:25.235896Z',
  creationSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
  lastChangeTimestamp: '2023-12-01T13:44:04.832072Z',
  lastChangeSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
};

export const expertReference: ExpertReference = {
  expertID: 'f016f204-c307-483b-8891-680a3c974c53',
  name: 'TEST MEMORI',
  description: 'lui √® competente',
  default: true,
  expertMemoriID: 'dfc44f1c-1ba7-4e1e-a234-28aa8b6b3d32',
  expertBaseURL: 'http://localhost:7778',
  creationTimestamp: '2023-12-01T13:40:25.235896Z',
  creationSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
  lastChangeTimestamp: '2023-12-01T13:44:04.832072Z',
  lastChangeSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
};

export const venue: Venue = {
  latitude: 44.66579,
  longitude: 11.48823,
  placeName: 'Altedo, Bologna, Italy',
  uncertainty: 2,
};

export const memoryQuestion: Memory = {
  memoryID: '08c0697d-f7f3-4a47-9970-aff75f01fb6c',
  memoryType: 'Question',
  lastRead: undefined,
  readOccurrences: undefined,
  receiverID: undefined,
  receiverTag: undefined,
  receiverName: undefined,
  media: [
    {
      mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
      mimeType: 'text/html',
      url: 'https://rawmaterial.it/it',
      title: 'RawMaterial',
    },
  ],
  title: 'sei umano',
  titleVariants: ['sei reale'],
  answers: [
    {
      text: 'se ti dico si ci credi?',
      preformatted: false,
      creationTimestamp: '2022-03-23T09:37:25.410377',
      creationName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
      lastChangeTimestamp: '2022-03-23T09:37:25.410406',
      lastChangeName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
    },
    {
      text: 'probabilmente lo sono pi√π di te',
      preformatted: false,
      creationTimestamp: '2022-03-23T09:37:36.957656',
      creationName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
      lastChangeTimestamp: '2022-03-23T09:37:36.957657',
      lastChangeName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
    },
  ],
  date: undefined,
  dateUncertaintyDays: undefined,
  placeName: undefined,
  placeLatitude: undefined,
  placeLongitude: undefined,
  placeUncertaintyKm: undefined,
  preformatted: false,
  conclusive: false,
  notPickable: false,
  contextVarsToSet: {
    SUBJECT: 'TRAVEL',
  },
  contextVarsToMatch: {
    SUBJECT: 'TRANSPORTS',
  },
  creationTimestamp: '2022-03-23T09:37:16.568149',
  creationName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
  lastChangeTimestamp: '2022-03-23T09:37:36.957695',
  lastChangeName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
};
