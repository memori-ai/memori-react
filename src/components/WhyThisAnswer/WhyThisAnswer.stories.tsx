import React from 'react';
import { Meta, Story } from '@storybook/react';
import { sessionID, memoryQuestion } from '../../mocks/data';
import I18nWrapper from '../../I18nWrapper';
import WhyThisAnswer, { Props } from './WhyThisAnswer';

import './WhyThisAnswer.css';
import { SearchMatches } from '@memori.ai/memori-api-client/dist/types';

const meta: Meta = {
  title: 'Why This Answer',
  component: WhyThisAnswer,
  argTypes: {
    visible: {
      control: {
        type: 'boolean',
      },
    },
  },
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<Props> = args => (
  <I18nWrapper>
    <WhyThisAnswer
      {...args}
      apiURL="https://backend.memori.ai"
      sessionID={sessionID}
      message={{
        questionAnswered: 'Test message',
        text: 'This is a test content',
        date: '2021-01-01',
        placeName: 'Test Place',
        placeLatitude: 0,
        placeLongitude: 0,
        placeUncertaintyKm: 0,
        contextVars: {
          KEY: 'value',
        },
      }}
      closeDrawer={() => {}}
    />
  </I18nWrapper>
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  visible: true,
};

export const Loading = Template.bind({});
Loading.args = {
  visible: true,
  _TEST_loading: true,
};

export const WithData = Template.bind({});
WithData.args = {
  visible: true,
  initialMatches: [
    {
      confidence: 0.8,
      confidenceLevel: 'HIGH',
      memory: {
        memoryID: '1',
        title: 'This is the title of the content',
        titleVariants: [
          "This is a variant of the content's title",
          'This is a test content',
        ],
        answers: [
          {
            text: 'This is a test answer',
          },
          {
            text: 'This is another answer',
          },
        ],
      },
    } as SearchMatches,
    {
      confidence: 0.5,
      confidenceLevel: 'LOW',
      memory: {
        memoryID: '2',
        title: 'Content with a long answer',
        titleVariants: undefined,
        answers: [
          {
            text: 'Suspendisse a sodales nulla, sed semper nisi. Suspendisse a sodales nulla, sed semper nisi. Suspendisse a sodales nulla, sed semper nisi. Suspendisse a sodales nulla, sed semper nisi. Suspendisse a sodales nulla, sed semper nisi.',
          },
        ],
      },
    } as SearchMatches,
    {
      confidence: 0.5,
      confidenceLevel: 'LOW',
      memory: {
        memoryID: '3',
        title: 'Content with sources',
        titleVariants: undefined,
        answers: [
          {
            text: 'This is a test answer',
          },
        ],
        media: [
          {
            mediumID: '1',
            mimeType: 'text/plain',
            content:
              'This is a source. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse a sodales nulla, sed semper nisi. Suspendisse a sodales nulla, sed semper nisi.',
          },
          {
            mediumID: '2',
            mimeType: 'text/plain',
            content:
              'This is a source.\n\nLorem ipsum dolor sit amet, consectetur adipiscing elit.\nCras lobortis volutpat nunc.\nProin tincidunt enim in felis aliquet, a ultricies purus bibendum.\n\nQuisque in ultrices lectus.\nNulla at urna diam.\n\nProin sodales lobortis libero eu facilisis.',
          },
        ],
      },
    } as SearchMatches,
    {
      confidence: 0.7,
      confidenceLevel: 'MEDIUM',
      memory: {
        ...memoryQuestion,
        memoryID: '4',
        title: 'Content with links',
        titleVariants: undefined,
        answers: [
          {
            text: 'This is a an answer',
          },
        ],
        media: [
          {
            mediumID: '1',
            mimeType: 'text/html',
            url: 'https://memori.ai',
            title: 'Memori.AI',
          },
          {
            mediumID: '2',
            mimeType: 'text/html',
            url: 'https://nzambello.dev',
            title: 'Nicola Zambello',
          },
        ],
      },
    } as SearchMatches,
  ],
};
