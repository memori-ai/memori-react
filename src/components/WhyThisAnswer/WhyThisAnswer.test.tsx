import React from 'react';
import { render } from '@testing-library/react';
import WhyThisAnswer from './WhyThisAnswer';
import { sessionID, memoryQuestion } from '../../mocks/data';

beforeEach(() => {
  // @ts-ignore
  window.IntersectionObserver = jest.fn(() => ({
    observe: jest.fn(),
    unobserve: jest.fn(),
    disconnect: jest.fn(),
    takeRecords: jest.fn(),
  }));
});

it('renders WhyThisAnswer hidden unchanged', () => {
  const { container } = render(
    <WhyThisAnswer
      apiURL="https://backend.memori.ai"
      sessionID={sessionID}
      visible={false}
      message={{
        questionAnswered: 'Test message',
        text: 'This is a test content',
        date: '2021-01-01',
        placeName: 'Test Place',
        placeLatitude: 0,
        placeLongitude: 0,
        placeUncertaintyKm: 0,
        contextVars: {
          KEY: 'value',
        },
      }}
      closeDrawer={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders WhyThisAnswer visible unchanged', () => {
  const { container } = render(
    <WhyThisAnswer
      apiURL="https://backend.memori.ai"
      sessionID={sessionID}
      visible={true}
      message={{
        questionAnswered: 'Test message',
        text: 'This is a test content',
        date: '2021-01-01',
        placeName: 'Test Place',
        placeLatitude: 0,
        placeLongitude: 0,
        placeUncertaintyKm: 0,
        contextVars: {
          KEY: 'value',
        },
      }}
      closeDrawer={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders WhyThisAnswer with data unchanged', () => {
  const { container } = render(
    <WhyThisAnswer
      apiURL="https://backend.memori.ai"
      sessionID={sessionID}
      visible={false}
      message={{
        questionAnswered: 'Test message',
        text: 'This is a test content',
        date: '2021-01-01',
        placeName: 'Test Place',
        placeLatitude: 0,
        placeLongitude: 0,
        placeUncertaintyKm: 0,
        contextVars: {
          KEY: 'value',
        },
      }}
      initialMatches={[
        {
          confidence: 0.8,
          confidenceLevel: 'HIGH',
          memory: {
            ...memoryQuestion,
            title: 'This is the title of the content',
            titleVariants: [
              "This is a variant of the content's title",
              'This is a test content',
            ],
            answers: [
              {
                text: 'This is a test answer',
              },
            ],
            memoryID: '1',
          },
        },
        {
          confidence: 0.7,
          confidenceLevel: 'MEDIUM',
          memory: {
            ...memoryQuestion,
            memoryID: '2',
            title: 'This is a test title',
            titleVariants: undefined,
            answers: [
              {
                text: 'This is a an answer',
              },
              {
                text: 'This is another answer',
              },
            ],
          },
        },
      ]}
      closeDrawer={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});
