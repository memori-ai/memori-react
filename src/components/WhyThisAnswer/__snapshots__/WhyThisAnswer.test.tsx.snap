// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders WhyThisAnswer hidden unchanged 1`] = `<div />`;

exports[`renders WhyThisAnswer visible unchanged 1`] = `
<div>
  <div
    style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
  />
</div>
`;

exports[`renders WhyThisAnswer with data unchanged 1`] = `<div />`;
